clang/lib/CodeGen/BackendUtil.cpp

```diff
@@ -1121,6 +1122,9 @@ void EmitAssemblyHelper::RunOptimizationPipeline(
       // Process properties and annotations
       MPM.addPass(CompileTimePropertiesPass());
 
+      // Remove SYCL metadata added by the frontend, like sycl_aspects
```

Comment:
Nit: `Remove sycl_aspects and sycl_types_that_use_aspects module-level metadata`

I think being specific is helpful in the comments. It might also make sense to elaborate on why we are removing them

Reply:
I think that it is also important to note that the pass should be the last in the pipeline. Could be helpful if someone attempts to add a pass which relies on the aforementioned metadata _after_ the cleanup pass

Reply:
> I think being specific is helpful in the comments. It might also make sense to elaborate on why we are removing them

TBH, like now the pass removes only module level metadata, but later I can image it will start removing more

> I think that it is also important to note that the pass should be the last in the pipeline.

Indeed, applied